src/nova-components/CreateFavorites/resources/js/components/TextInput.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <input
+    :type="type"
+    :class="{
+      'input-error': error || validationError || sameError,
+      'input-decoration': true,
+    }"
+    :placeholder="placeholder"
+    v-model="input"
+    @input="$emit('changeInputField', $data.input)"
```

Comment:
don't use events pass function as a prop to component then call this function inside component